integration/utils/utils.go

```diff
@@ -0,0 +1,217 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"k8s.io/api/core/v1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/crdstorage"
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	"github.com/giantswarm/ipam"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/microstorage"
+	"github.com/giantswarm/operatorkit/client/k8scrdclient"
+	"github.com/giantswarm/rangepool"
+)
+
+const (
+	vniMin      = 1
+	vniMax      = 1000
+	nodePortMin = 30100
+	nodePortMax = 31500
+
+	flannelCidrSize        = 26
+	flannelE2eNetworkRange = "10.1.0.0/16"
+
+	vniRangepoolNamespace     = "vni"
+	ingressRangepoolNamespace = "ingress"
+
+	giantswarmNamespace = "giantswarm"
+)
+
+func InitCRDStorage(h *framework.Host, l micrologger.Logger) (microstorage.Storage, error) {
+	var err error
+
+	k8sExtClient, err := apiextensionsclient.NewForConfig(h.RestConfig())
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var k8sCrdClient *k8scrdclient.CRDClient
+	{
+		var k8sCrdClientConfig k8scrdclient.Config
+		k8sCrdClientConfig.Logger = l
+		k8sCrdClientConfig.K8sExtClient = k8sExtClient
+
+		k8sCrdClient, err = k8scrdclient.New(k8sCrdClientConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	c := crdstorage.DefaultConfig()
+	c.CRDClient = k8sCrdClient
+	c.G8sClient = h.G8sClient()
+	c.K8sClient = h.K8sClient()
+	c.Logger = l
+
+	c.Name = "kvm-e2e"
+	c.Namespace = &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: giantswarmNamespace,
+		},
+	}
+
+	crdStorage, err := crdstorage.New(c)
+
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	l.Log("info", "booting crdstorage")
+	err = crdStorage.Boot(context.Background())
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return crdStorage, nil
+}
+
+func InitRangePool(crdStorage microstorage.Storage, l micrologger.Logger) (*rangepool.Service, error) {
+
+	rangePoolConfig := rangepool.DefaultConfig()
+	rangePoolConfig.Logger = l
+	rangePoolConfig.Storage = crdStorage
+
+	rangePool, err := rangepool.New(rangePoolConfig)
+	if err != nil {
+		return nil, microerror.Mask(err)
+
+	}
+
+	return rangePool, nil
+}
+
+func GenerateVNI(rangePool *rangepool.Service, clusterID string) (int, error) {
+	items, err := rangePool.Create(
+		context.Background(),
+		vniRangepoolNamespace,
+		rangePoolVNIID(clusterID),
+		1, // num
+		vniMin,
+		vniMax,
+	)
+
+	if err != nil {
+		return 0, microerror.Mask(err)
+	}
+
+	if len(items) != 1 {
+		return 0, microerror.Maskf(executionFailedError, "range pool VNI generation failed, expected 1 got %d", len(items))
+	}
+
+	return items[0], nil
+}
+
+func DeleteVNI(rangePool *rangepool.Service, clusterID string) error {
+	return rangePool.Delete(context.Background(), vniRangepoolNamespace, rangePoolVNIID(clusterID))
```

Comment:
Lets please always make the error handling explicit so the story the stack tells you in error cases reflects reality.

Reply:
I think it would be good to propagate a context here. We can create one when the actual test starts and dispatch it through. So we do not have to create one on each new level, because this is rather useless.